ruby/ql/lib/codeql/ruby/frameworks/GraphQL.qll

```diff
@@ -0,0 +1,439 @@
+  private GraphqlResolvableClass resolvableClass;
+
+  GraphqlResolveMethod() { this = resolvableClass.getMethod("resolve") }
+
+  override Parameter getARoutedParameter() { result = this.getAParameter() }
+
+  override string getFramework() { result = "GraphQL" }
+
+  /** Gets the mutation class containing this method. */
+  GraphqlResolvableClass getMutationClass() { result = resolvableClass }
+}
+
+/**
+ * A `load_*` method on a sub-class of either
+ * `GraphQL::Schema::RelayClassicMutation` or
+ * `GraphQL::Schema::Resolver`.
+ *
+ * This method takes user input (some kind of ID or specifier) and is intended
+ * to resolve the domain object using that ID.
+ *
+ * For example:
+ *
+ * ```rb
+ * module Mutation
+ *   class NameAnInstrument < BaseMutationn
+ *     argument :instrument_uuid, Types::Uuid,
+ *              required: true,
+ *              loads: ::Instrument,
+ *              as: :instrument,
+ *     argument :name, String, required: true
+ *
+ *     # GraphqlLoadMethod
+ *     def load_instrument(uuid)
+ *       ::Instrument.find_by(uuid: uuid)
+ *     end
+ *
+ *     def resolve(instrument:, name:)
+ *       instrument.set_name(name)
+ *     end
+ *   end
+ * end
+ * ```
+ */
+class GraphqlLoadMethod extends Method, HTTP::Server::RequestHandler::Range {
+  private GraphqlResolvableClass resolvableClass;
+
+  GraphqlLoadMethod() {
+    this.getEnclosingModule() = resolvableClass and
+    this.getName().regexpMatch("^load_.*")
+  }
+
+  override Parameter getARoutedParameter() { result = this.getAParameter() }
+
+  override string getFramework() { result = "GraphQL" }
+
+  /** Gets the mutation class containing this method. */
+  GraphqlResolvableClass getMutationClass() { result = resolvableClass }
+}
+
+/**
+ * A `ClassDeclaration` for a class that extends
+ * `GraphQL::Schema::Object`.
+ *
+ * These classes can contain `field` definitions for the graphql schema.
+ *
+ * For example:
+ *
+ * ```rb
+ * class Types::User < GraphQL::Schema::Object
+ *   field :email, String
+ * end
+ * ```
+ */
+class GraphqlFieldableClass extends ClassDeclaration {
+  GraphqlFieldableClass() { this instanceof GraphqlSchemaObjectClass }
+
+  GraphqlFieldDefinitionMethodCall getAFieldDefinitionMethodCall() {
+    result.(GraphqlFieldDefinitionMethodCall).getReceiverClass() = this
+  }
+}
+
+/**
+ * A `MethodCall` that represents calling a class method on a
+ * "fieldable" class (a sub-class of `GraphQL::Schema::Object`)
+ */
+class GraphqlFieldableClassMethodCall extends MethodCall {
+  private GraphqlFieldableClass recvCls;
+
+  GraphqlFieldableClassMethodCall() {
+    // e.g. Foo.some_method(...)
+    recvCls.getModule() = resolveScopeExpr(this.getReceiver())
+    or
+    // e.g. self.some_method(...) within a graphql Object or Interface
+    this.getReceiver() instanceof Self and
+    this.getEnclosingModule() = recvCls
+  }
+
+  /** The `GraphqlFieldableClass` representing the receiver of this method. */
```

Comment:
There's a convention (for some reason) that getters like this have a doc comment that starts `Gets the ...`. Unfortunately I think the CI job that catches this doesn't run on external PRs. We will want to make the same change to the other cases in this file.

```suggestion
  /** Gets the `GraphqlFieldableClass` representing the receiver of this method. */
```